test/puppetlabs/trapperkeeper/services_test.clj

```diff
@@ -62,6 +64,67 @@
           app (bootstrap-services-with-empty-config [service1])]
       (is (not (nil? app))))))
 
+(deftest test-restart-file-correct-count
+  (testing "create a restart file and check that it correctly increments when a service starts"
+    (let [service1 (service Service1
+                            []
+                            (service1-fn [this] "hi"))
+          temp-file (kitchensink/temp-file-name "counter")]
```

Comment:
hrm.  We should probably look into that... but it's not a blocker for this PR, obviously.  And `temp-file-name` seems fine for your use case.